onboarding_documentation/technical_documentation/metamist_docker/metamist_docker_tutorial.md

```diff
@@ -0,0 +1,737 @@
+
+- Tips:
+  - Create a new job called `j2` using `j2 = b.new_python_job()` (make sure to give the job a name inside the brackets)
+  - The new job needs an image that has python, we can just use the `driver_image` for this: `j2.image(get_config()['workflow']['driver_image'])`
+  - We need to tell Hail that this second job is dependent on the first job, we can do this using `j2.depends_on(j)`
+
+<br>
+<br>
+
+<details>
+<summary>Click to reveal full code</summary>
+
+```python
+def main(project: str, sgids: list[str]):
+    # region: metadata queries
+    # This section is all executed prior to the workflow being scheduled,
+    # so we have access to all variables
+
+    # Metamist query for files
+    file_dict = get_assays(project, sgids)
+    # endregion
+
+    # region: Batch time
+    b = get_batch('Schedule some worker tasks')
+    for sg, files in file_dict.items():
+        # check that we have 2 files (assuming FQ, rather than BAM)
+        assert len(files) == 2
+
+        # Create a job for each sample
+        j = b.new_job('FastQE', {'tool': 'fastqe'})
+
+        # Set the docker image to use in this job
+        # this pulls the image path from the portion of the config
+        # populated by the images repository
+        j.image(image_path('fastqe'))
+        j.storage('1Gi')
+
+        # read data into the batch tmp resource location
+        file_1 = b.read_input(files[0])
+        file_2 = b.read_input(files[1])
+
+        # Set the command to run
+        # batch.read_input will create a new path like /io/batch/75264c/CPGAAAA_1.fq.gz
+        # accessible from inside the job container, and unique to this batch/job
+        cmd = f'fastqe {file_1} {file_2} --html > {j.out_fastqe}'
+        j.command(
+            # f'echo "Hello world, I am a job for {sg}!, using {file_1} and {file_2}"'
+            # f'I\'m also creating an output file at {j.output}'
+            # f'echo "Some outputs" > {j.output}'
+            command(cmd)
+        )
+        j2 = b.new_python_job(f'Register analysis output for {sg}')
+        j2.image(get_config()['workflow']['driver_image'])
+        j2.call(create_analysis_entry, sg)
+        j2.depends_on(j)
+        # read the output out into GCP
+        # The helper method output_path() will create a new path based on the current project,
+        # test/main, and the output prefix provided to analysis_runner
+        # -o my_output
+        # --dataset my-dataset
+        # --access_level test
+        # output_path('this_file.txt')
+        # -> gs://cpg-my-dataset-test/my_output/this_file.txt
+        b.write_output(
+            j.out_fastqe, output_path(f'{sg}.html', category='web')
+        )  # category='web' writes it to the web bucket
+    b.run(wait=False)
+```
+
+</details>
+<br>
+
+
+Putting it all together, our python script should look like this:
```

Comment:
As previously, it makes sense to explain what this script would do and link to it, rather than recreating a full script in the documentation